src/main/java/seedu/address/logic/commands/UndoableCommand.java

```diff
@@ -5,24 +5,20 @@
 import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
 
 import seedu.address.logic.commands.exceptions.CommandException;
-import seedu.address.model.EventPlanner;
-import seedu.address.model.ReadOnlyEventPlanner;
 
 /**
  * Represents a command which can be undone and redone.
  */
 public abstract class UndoableCommand extends Command {
-    private ReadOnlyEventPlanner previousAddressBook;
-
-    protected abstract CommandResult executeUndoableCommand() throws CommandException;
 
+    // @@author bayweiheng
     /**
-     * Stores the current state of {@code model#addressBook}.
+     * Represents the Command to be executed to revert the changes made by the UndoableCommand.
+     * We mandate that the oppositeCommand of an UndoableCommand be undoable as well
      */
-    private void saveAddressBookSnapshot() {
-        requireNonNull(model);
-        this.previousAddressBook = new EventPlanner(model.getEventPlanner());
-    }
+    protected UndoableCommand oppositeCommand;
```

Comment:
I think that `reverseCommand` or `inverseCommand` is a better name

Reply:
Did think of that, but I think opposite is easier to read/understand without loss in clarity